### PR TITLE
Feature/styled components stiches migration

### PR DESCRIPTION
**Pull request by apoorva-srivastav**
styled-components library is now replaced with nextui library 
The nextui default theme can be overriden by a custom theme in the root of the application.
Components can be created either by using nextui components or custom components which are styled.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "Accelerator"
   ],
   "dependencies": {
+    "@stitches/react": "^1.2.7",
     "axios": "^0.26.0",
```
Comment (khus29) [inline package.json]: I believe we don't need to add dependency here. Please validate n remove.

File: templates/common/config/stitches.config.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { globalTheme } from 'themes';
+
+export const {
+  styled,
+  css,
+  globalCss,
+  keyframes,
+  getCssText,
+  theme,
+  createTheme,
```
Comment (khus29) [inline templates/common/config/stitches.config.ts]: As discussed please add other configs just for demonstration.

File: templates/common/storybook/.storybook/preview.js [hunk 3]
```diff
@@ -14,10 +11,6 @@ export const parameters = {
   }
 };
 
-addDecorator((storyFn, context) => (
```
Comment (vinodloha) [inline templates/common/storybook/.storybook/preview.js:17]: How will the theme be available for every component if we don't have this?

File: templates/common/src/components/atoms/Button/tests/Button.test.tsx [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
 import { render, screen } from '@testing-library/react';
-import { ThemeProvider } from 'styled-components';
-import { theme } from 'themes';
 
 import Button from '../index';
 
 describe('<Button />', () => {
   beforeEach(() => {
     render(
-      <ThemeProvider theme={theme}>
```
Comment (vinodloha) [inline templates/common/src/components/atoms/Button/tests/Button.test.tsx:11]: Good change.

File: templates/common/src/components/atoms/Button/Button.tsx [hunk 5]
```diff
@@ -7,10 +7,9 @@
 
 // Node_Module Imports
 import React from 'react';
-import styled from 'styled-components';
 
 // Relative imports
-import styles, { AppContainer } from './Button.style';
+import { AppContainer } from './Button.style';
```
Comment (vinodloha) [inline templates/common/src/components/atoms/Button/Button.tsx]: Lets call it CompRoot instead of AppContainer

File: templates/common/packages/themes/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "themes",
-  "version": "0.1.0",
-  "private": true,
-  "main": "./theme.ts",
-  "license": "MIT"
-}
+    "name": "themes",
```
Comment (vinodloha) [inline templates/common/packages/themes/package.json]: I think you need to fix your editor configurations to avoid some of these unnecessary formatting changes. We have necessary extensions available in vs-code extension recommendations

File: templates/common/config/stitches.config.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { createStitches } from '@stitches/react';
+import { customTheme } from '../src/theme/theme';
+import { globalTheme } from 'themes';
+
+export const {
```
Comment (vinodloha) [inline templates/common/config/stitches.config.ts]: I would recommend we manage everything related to theme within the theme package. There is no need to create multiple themes. We just need to demonstrate how to extend the one you get from https://nextui.org/

File: templates/base/src/pages/_app.tsx [hunk 8]
```diff
@@ -1,27 +1,31 @@
...
 import GlobalContextProvider from '../stores/globalContext';
+import { globalCss } from '../../config/stitches.config';
 
 // Will be called once for every metric that has to be reported.
 // export function reportWebVitals(metric: any) {
 // These metrics can be sent to any analytics service
 // console.log(metric);
 // }
 
+const globalStyles = globalCss({
```
Comment (vinodloha) [inline templates/base/src/pages/_app.tsx]: This should be coming from theme, if at all needed.

**Review by khus29 — DISMISSED** [review]

**Review by vinodloha — APPROVED** [review]
